### DISCUSSION TITLE
How to render edge text at the beginning or end of the edge path using `<EdgeLabelRenderer>` ?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [xyflow/xyflow#2863] [Q&A]
Hi I am a reactflow pro subscriber 🙏 

### Question
How do I render the text label for an edge at the beginning of the edge instead of the middle using [`<EdgeLabelRenderer />`](https://reactflow.dev/docs/api/edges/edge-label-renderer/).❓ 
I had trouble finding examples in the docs for how to place the label at the beginning or end of an edge using [`<EdgeLabelRenderer />`](https://reactflow.dev/docs/api/edges/edge-label-renderer/).

Docs example: https://reactflow.dev/docs/examples/edges/edge-label-renderer/
![CleanShot 2023-02-25 at 19 14 16@2x](https://user-images.githubusercontent.com/6743796/221390323-4652b005-f78f-4824-b022-209c6eab661e.png)

### End Goal:
I'd like to to render nodes that connect to each other and have their edge text at the beginning or end like this:
![CleanShot 2023-02-25 at 19 52 09](https://user-images.githubusercontent.com/6743796/221391429-b7ac8e4a-e81a-45bb-939c-3eb38a2453c2.png)

### What I've attempted
Currently I have this stackblitz setup with some starter code using `<EdgeLabelRenderer>`
Having trouble finding documentation on how to move the label text to beginning or end & learning which edge util function might be the one to use

Stackblitz Code:  https://stackblitz.com/edit/react-ts-ehx5sg?file=App.tsx
![CleanShot 2023-02-25 at 20 02 04@2x](https://user-images.githubusercontent.com/6743796/221391650-e08c7b0b-fc66-4008-837e-440137336daa.png)

## Related
- https://github.com/wbkd/react-flow/issues/2864
- https://discord.com/channels/771389069270712320/1046759869409140748/1046759872907182160

**Comment by moklick** [comment]
Hey! You can use `sourceX`, `sourceY` and `targetX`, `targetY` to achieve this. Example with source: https://stackblitz.com/edit/react-ts-j2el6a?file=components%2FCustomEdge.tsx
